server/src/main/java/org/opensearch/index/IndexService.java

```diff
@@ -484,11 +491,31 @@ public synchronized IndexShard createShard(
                     warmer.warm(reader, shard, IndexService.this.indexSettings);
                 }
             };
-
+            DiscoveryNode node = targetNode;
             Store remoteStore = null;
-            if (this.indexSettings.isRemoteStoreEnabled()) {
-                Directory remoteDirectory = remoteDirectoryFactory.newDirectory(this.indexSettings, path);
-                remoteStore = new Store(shardId, this.indexSettings, remoteDirectory, lock, Store.OnClose.EMPTY, path);
+            boolean seedRemote = false;
+            if (node.isRemoteStoreNode()) {
+                if (this.indexSettings.isRemoteStoreEnabled()) {
+                    Directory remoteDirectory = remoteDirectoryFactory.newDirectory(this.indexSettings, path);
+                    remoteStore = new Store(shardId, this.indexSettings, remoteDirectory, lock, Store.OnClose.EMPTY, path);
+                } else {
+                    logger.info("Hitting the migration case");
+                    if (sourceNode != null && sourceNode.isRemoteStoreNode() == false) {
+                        assert routing.primary();
+                        logger.info("Hitting the migration case where we need to seed remote store");
+                        seedRemote = true;
+                    }
+                    RemoteSegmentStoreDirectoryFactory directoryFactory = new RemoteSegmentStoreDirectoryFactory(
```

Comment:
Why can't we use `remoteDirectoryFactory` that is already available with IndexService?

Reply:
That is because it uses `indexSettings` to figure out the remote segment repository .

Reply:
At this stage, indexSettings are not having the repository information, is it?

Reply:
Can we evaluate setting the repository settings upfront? The replication type setting can be changed at the very end after the migration is done.

Reply:
Also with https://github.com/opensearch-project/OpenSearch/issues/12608, we would need to store information about the path in the index metadata upfront itself as well fyi.

Reply:
> Can we evaluate setting the repository settings upfront? The replication type setting can be changed at the very end after the migration is done.

We use repository settings as a way to determine whether the index is remote or not . Also this is evaluation is on a shard level . We don't want to change that based on one shard's migration .

Reply:
created https://github.com/opensearch-project/OpenSearch/issues/12688

Reply:
> That is because it uses indexSettings to figure out the remote segment repository.

`IndexSettings` is checked to create directory from directory factory but the factory is already available, right? Can we use the same factory?